include/net/buf.h

```diff
@@ -590,7 +601,9 @@ struct net_buf *net_buf_get(struct k_fifo *fifo, s32_t timeout);
  */
 static inline void net_buf_destroy(struct net_buf *buf)
 {
-	k_lifo_put(&buf->pool->free, buf);
+	struct net_buf_pool *pool = net_buf_pool_get(buf->pool_id);
```

Comment:
No other reason except code clarity. Presumably the compiler will generate the same code either way with the default optimization level (I'd need to verify that however).